update-engine/src/events.rs

```diff
@@ -0,0 +1,688 @@
+                      },
+                      "attempt_elapsed": {
+                        "secs": 0,
+                        "nanos": 0
+                      }
+                    }
+                  }
+                "#,
+                StepEvent {
+                    total_elapsed: Duration::ZERO,
+                    kind: StepEventKind::Unknown,
+                },
+            ),
+            (
+                r#"
+                  {
+                    "total_elapsed": {
+                      "secs": 0,
+                      "nanos": 0
+                    },
+                    "data": {
+                      "kind": "execution_completed",
+                      "last_step": {
+                        "info": {
+                          "id": 0,
+                          "component": "foo",
+                          "description": "Description",
+                          "index": 0,
+                          "component_index": 0,
+                          "total_component_steps": 1
+                        },
+                        "metadata": null
+                      },
+                      "last_attempt": 1,
+                      "last_outcome": {
+                        "kind": "success",
+                        "metadata": null
+                      },
+                      "step_elapsed": {
+                        "secs": 0,
+                        "nanos": 0
+                      },
+                      "attempt_elapsed": {
+                        "secs": 0,
+                        "nanos": 0
+                      },
+                      "unknown_field": 123
+                    }
+                  }
+                "#,
+                StepEvent::<TestSpec> {
+                    total_elapsed: Duration::ZERO,
+                    kind: StepEventKind::ExecutionCompleted {
+                        last_step: StepInfoWithMetadata {
+                            info: StepInfo {
+                                id: 0,
+                                component: "foo".to_owned(),
+                                description: "Description".into(),
+                                index: 0,
+                                component_index: 0,
+                                total_component_steps: 1,
+                            },
+                            metadata: None,
+                        },
+                        last_attempt: 1,
+                        last_outcome: StepOutcome::Success {
+                            metadata: serde_json::Value::Null,
+                        },
+                        step_elapsed: Duration::ZERO,
+                        attempt_elapsed: Duration::ZERO,
+                    },
+                },
+            ),
+        ];
+
+        for (index, (input, expected)) in tests.into_iter().enumerate() {
+            // let serialized = serde_json::to_string_pretty(&expected).unwrap();
+            // println!("{serialized}");
```

Comment:
Maybe remove - leftover from debugging?

Reply:
I made it so that it's now logged to a file while tests are running. it's quite useful to have a way to examine serialized metadata, and I figured this was an easy spot.